tests/EdgeToEdge/Routes/ShowDonationConfirmationRouteTest.php

```diff
@@ -23,24 +24,23 @@ class ShowDonationConfirmationRouteTest extends WebRouteTestCase {
 	private const ACCESS_DENIED_TEXT = 'access_denied_donation_confirmation';
 
 	private Donation $donation;
+	private Payment $payment;
 
 	public function testGivenValidRequest_confirmationPageContainsDonationData(): void {
-		$this->markTestIncomplete( "This should work again when we finish updating the membership controllers" );
-
 		$this->modifyEnvironment( function ( FunFunFactory $factory ): void {
+			$this->payment = $this->newStoredPayment( ValidPayments::newDirectDebitPayment(), $factory );
```

Comment:
I'd put the code for storing the the payment in the `newStoredDonation` method (giving it a `Payment` parameter and modifying it). You can then omit setting the token generator in the `newStoredPayment`. This will make the tests shorter and more readable overall.